api/views/workflow.py

```diff
@@ -49,12 +50,21 @@ class SourceViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = SourceSerializer
     permission_classes = [DjangoModelPermissionsOrAnonReadOnly, ]
 
-    queryset = Source.objects.none()  # Required for DjangoModelPermissions
+    queryset = Source.objects.exclude(icon='').exclude(is_deleted=True)  # Required for DjangoModelPermissions
 
     VALID_IMAGE_TYPES = ('image/png', 'image/jpeg')
 
     def get_queryset(self):
-        return Source.objects.exclude(icon='').exclude(is_deleted=True)
+        queryset = self.queryset
+        source_names = self.request.query_params.get('long_titles', None)
+
+        if source_names is None:
+            return queryset
+        source_names = source_names.split(',')
```

Comment:
>OpenKnowledge@NAU - Search the scholarly, creative and instructional output of NAU
University of Minnesota, Digital Conservancy
University of Illinois at Urbana-Champaign, IDEALS